examples/10_streaming_write.py

```diff
@@ -3,31 +3,92 @@
 import numpy as np
 import sys
 
-if __name__ == "__main__":
-    if 'adios2' not in io.variants or not io.variants['adios2']:
-        print('This example requires ADIOS2')
-        sys.exit(0)
-
-    series = io.Series("stream.sst", io.Access_Type.create)
+# pass-through for ADIOS2 engine parameters
+# https://adios2.readthedocs.io/en/latest/engines/engines.html
+config = """
```

Comment:
Let's replace this with the [import json](https://docs.python.org/3/library/json.html) objects because it's more readable and a standard lib in Python :)

Reply:
I think, in that case we should make it possible to write something like:
```python
series = io.Series("asdf.bp", io.Access.create, {"adios2": {"engine": {"type": "bp4"}}}
```
That's currently not possible because the `Series` constructor accepts strings only, so a string conversion needs to happen inside Python.

Reply:
That says the same - we populate a json object and then pass the string representation of it to `io.Series`:
```py
import json
import collections

config = {'adios2': {'engine': {}, 'dataset': {}}}
config['adios2']['engine'] = {'parameters': {'Threads': '4'}}
config['adios2']['dataset'] = {'operators': {'type': 'bzip2'}}

series = io.Series("stream.sst", io.Access_Type.create, json.dumps(config))
```

Reply:
What I was trying to say is that it would just be a bit neater to avoid the `json.dumps()` and make it look like we understand the JSON value natively. But it might be too much hassle.

Reply:
I've used your first suggestion now

Reply:
Cool, thx! I think that's good for the current PR :)

> What I was trying to say is that it would just be a bit neater to avoid the json.dumps() and make it look like we understand the JSON value natively. But it might be too much hassle.

We can also add a new overload in Python that understands dictionaries and do `json.dumps(config_dict)` internally :)